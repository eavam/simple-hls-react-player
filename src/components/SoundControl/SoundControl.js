// @flow
import React from 'react';
import styled from 'react-emotion';
import withConnect from '../../hocs/withConnect';
import { baseIcon } from '../styledComponens';
import { SoundIcon, OffIcon } from '../svg-icons';

import { handleChangeVolume } from '../../actions/PlayerActions';
import { toggleMute } from '../../actionTypes';

const Root = styled('div')`
  display: flex;
  align-items: center;
`;
const VolumeControl = styled('div')`
  position: relative;
  cursor: pointer;
  height: 5px;
  width: 60px;
  background-color: rgba(255, 255, 255, 0.5);
  box-shadow: inset ${props => (props.muted ? 0 : props.volume)}px 0px 0 #fff;
`;

type Props = {
  volume?: number,
  muted?: boolean,
  onClick?: () => void,
  onMouseDown?: () => void,
};

export const SoundControl = ({ volume, muted, onClick, onMouseDown }: Props) => (
  <Root>
    <div className={baseIcon} onClick={onClick} role="presentation">
      {muted ? <OffIcon /> : <SoundIcon />}
    </div>
    <VolumeControl onMouseDown={onMouseDown} muted={muted} volume={volume} />
  </Root>
);

SoundControl.defaultProps = {
  volume: 0,
  muted: false,
  onClick: () => {},
  onMouseDown: () => {},
};

export default withConnect(['volume', 'muted'], {
  onClick: toggleMute,
  onMouseDown: handleChangeVolume,
})(SoundControl);
