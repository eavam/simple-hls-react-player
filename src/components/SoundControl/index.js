import React from 'react';
import PropTypes from 'prop-types';
import styled from 'react-emotion';
import { baseIcon } from '../styledComponens';
import { SoundIcon, OffIcon } from '../svg-icons';

const SoundControl = ({ volume, muted, onClick, onMouseDown }) => (
  <Root>
    <div className={baseIcon} onClick={onClick} role="presentation">
      {muted ? <OffIcon /> : <SoundIcon />}
    </div>
    <VolumeControl onMouseDown={onMouseDown} muted={muted} volume={volume} />
  </Root>
);

SoundControl.propTypes = {
  volume: PropTypes.number,
  muted: PropTypes.bool,
  onClick: PropTypes.func,
  onMouseDown: PropTypes.func
};

SoundControl.defaultProps = {
  volume: 0,
  muted: false,
  onClick: () => {},
  onMouseDown: () => {}
};

const Root = styled('div')`
  display: flex;
  align-items: center;
`;
const VolumeControl = styled('div')`
  position: relative;
  cursor: pointer;
  height: 5px;
  width: 60px;
  background-color: rgba(255, 255, 255, 0.5);
  box-shadow: inset ${props => (props.muted ? 0 : props.volume)}px 0px 0 #fff;
`;

export default SoundControl;
