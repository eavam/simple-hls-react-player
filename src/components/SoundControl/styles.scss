.sound-control {
  display: flex;
  align-items: center;
}

.volume-control {
  position: relative;
  cursor: pointer;
  height: 5px;
  width: 60px;
  background-color: rgba(#fff, 0.5);

}