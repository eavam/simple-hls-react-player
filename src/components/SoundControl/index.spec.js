import React from 'react'
import renderer from 'react-test-renderer'
import SoundControl from './index.js'

test('SoundControl', () => {

  const props = {
    volume: '34',
    muted: false,
    toggleMute: () => {},
    handleChangeVolume: () => {},
  }


  let component = renderer.create(
    <SoundControl {...props} />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()

  props.muted = true

  component = renderer.create(
    <SoundControl {...props} />
  )
  tree = component.toJSON()
  expect(tree).toMatchSnapshot()

})