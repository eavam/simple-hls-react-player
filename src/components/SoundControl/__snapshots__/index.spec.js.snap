// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SoundControl default 1`] = `
<div
  className="css-1sg2lsz"
>
  <div
    className="css-1htvj9q"
    onClick={[Function]}
    role="presentation"
  >
    <svg
      viewBox="0 0 17.45 24.61"
    >
      <path
        d="M16.52.17a1.64 1.64 0 0 0-1.73.18L9.52 4.48a3.2 3.2 0 0 0-.9-.13H3.21A3.21 3.21 0 0 0 0 7.56v9.36a3.21 3.21 0 0 0 3.21 3.21h5.41a3.22 3.22 0 0 0 .78-.1l5.39 4.23A1.64 1.64 0 0 0 17.45 23V1.64a1.64 1.64 0 0 0-.93-1.47zm0 0"
        fill="#fff"
      />
    </svg>
  </div>
  <div
    className="css-hvewp1"
    muted={false}
    onMouseDown={[Function]}
  />
</div>
`;

exports[`SoundControl muted false 1`] = `
<div
  className="css-1sg2lsz"
>
  <div
    className="css-1htvj9q"
    onClick={[Function]}
    role="presentation"
  >
    <svg
      viewBox="0 0 17.45 24.61"
    >
      <path
        d="M16.52.17a1.64 1.64 0 0 0-1.73.18L9.52 4.48a3.2 3.2 0 0 0-.9-.13H3.21A3.21 3.21 0 0 0 0 7.56v9.36a3.21 3.21 0 0 0 3.21 3.21h5.41a3.22 3.22 0 0 0 .78-.1l5.39 4.23A1.64 1.64 0 0 0 17.45 23V1.64a1.64 1.64 0 0 0-.93-1.47zm0 0"
        fill="#fff"
      />
    </svg>
  </div>
  <div
    className="css-hvewp1"
    muted={false}
    onMouseDown={[Function]}
  />
</div>
`;

exports[`SoundControl volume 1`] = `
<div
  className="css-1sg2lsz"
>
  <div
    className="css-1htvj9q"
    onClick={[Function]}
    role="presentation"
  >
    <svg
      viewBox="0 0 17.45 24.61"
    >
      <path
        d="M16.52.17a1.64 1.64 0 0 0-1.73.18L9.52 4.48a3.2 3.2 0 0 0-.9-.13H3.21A3.21 3.21 0 0 0 0 7.56v9.36a3.21 3.21 0 0 0 3.21 3.21h5.41a3.22 3.22 0 0 0 .78-.1l5.39 4.23A1.64 1.64 0 0 0 17.45 23V1.64a1.64 1.64 0 0 0-.93-1.47zm0 0"
        fill="#fff"
      />
    </svg>
  </div>
  <div
    className="css-1m4e3bc"
    muted={false}
    onMouseDown={[Function]}
  />
</div>
`;
